eth/backend.go

```diff
@@ -296,7 +299,11 @@ func New(ctx *node.ServiceContext, config *Config) (*Ethereum, error) {
 	newPool := core.NewTxPool(eth.chainConfig, eth.EventMux(), eth.blockchain.State, eth.blockchain.GasLimit)
 	eth.txPool = newPool
 
-	if eth.protocolManager, err = NewProtocolManager(eth.chainConfig, config.FastSync, config.NetworkId, eth.eventMux, eth.txPool, eth.pow, eth.blockchain, chainDb); err != nil {
+	m := downloader.FullSync
+	if config.FastSync {
+		m = downloader.FastSync
+	}
+	if eth.protocolManager, err = NewProtocolManager(eth.chainConfig, m, uint64(config.NetworkId), eth.eventMux, eth.txPool, eth.pow, eth.blockchain, chainDb); err != nil {
```

Comment:
`ProtocolManager` gets an actual `downloader` _mode_ (`Full/FastSync`) instead of a bool. There is also a thing with casting the network id but AFAIR that was just cuz some more or less arbitrary refactoring on the handler side.